NaturalOrderComparator.java

```diff
@@ -82,67 +74,51 @@ public int compare(Object o1, Object o2)
         char ca, cb;
         int result;
 
-        while (true)
-        {
-            // only count the number of zeroes leading the last number compared
+        while (true) {
+            // Only count the number of zeroes leading the last number compared
             nza = nzb = 0;
 
             ca = charAt(a, ia);
             cb = charAt(b, ib);
 
             // skip over leading spaces or zeros
-            while (Character.isSpaceChar(ca) || ca == '0')
-            {
-                if (ca == '0')
-                {
+            while (Character.isSpaceChar(ca) || ca == '0') {
+                if (ca == '0') {
                     nza++;
-                }
-                else
-                {
-                    // only count consecutive zeroes
+                } else {
+                    // Only count consecutive zeroes
                     nza = 0;
                 }
 
                 ca = charAt(a, ++ia);
             }
 
-            while (Character.isSpaceChar(cb) || cb == '0')
-            {
-                if (cb == '0')
-                {
+            while (Character.isSpaceChar(cb) || cb == '0') {
+                if (cb == '0') {
                     nzb++;
-                }
-                else
-                {
-                    // only count consecutive zeroes
+                } else {
+                    // Only count consecutive zeroes
                     nzb = 0;
                 }
 
                 cb = charAt(b, ++ib);
             }
 
-            // process run of digits
-            if (Character.isDigit(ca) && Character.isDigit(cb))
-            {
-                if ((result = compareRight(a.substring(ia), b.substring(ib))) != 0)
-                {
-                    return result;
-                }
+            // Process run of digits
+            if (Character.isDigit(ca) && Character.isDigit(cb) && ((result = compareRight(a.substring(ia), b.substring(ib))) != 0)) {
+                return result;
```

Comment:
I agree this doesn't change the semantic of the code, but elsewhere you have kept the nested ifs, why not here too? And all the more since it's not a pure test, but also setting a variable…